adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsInterfaceGenerator.java

```diff
@@ -0,0 +1,262 @@
+        final Map<String, UriParameter> pathParams = action.getResource().getUriParameters();
+        final AnnotationSpec actionTypeAnnotation;
+
+        if (actionType == GET) {
+            actionTypeAnnotation = AnnotationSpec.builder(javax.ws.rs.GET.class).build();
+        } else if (actionType == POST) {
+            actionTypeAnnotation = AnnotationSpec.builder(javax.ws.rs.POST.class).build();
+        } else {
+            throw new IllegalStateException(format("Unsupported action type %s", actionType));
+        }
+
+        return methodBuilder(resourceMethodName)
+                .addModifiers(PUBLIC, ABSTRACT)
+                .addAnnotation(actionTypeAnnotation)
+                .addAnnotations(annotationsForProduces(responseMimeTypes))
+                .addParameters(methodPathParams(pathParams.keySet()))
+                .addParameters(methodQueryParams(queryParams.keySet()))
+                .returns(Response.class);
+    }
+
+    /**
+     * Generate code to add all query parameters to the params map.
+     *
+     * @param queryParams the query param names to add to the map
+     * @return the {@link CodeBlock} that represents the generated code
+     */
+    private List<ParameterSpec> methodQueryParams(final Set<String> queryParams) {
+        return methodParams(queryParams, QueryParam.class);
+    }
+
+    /**
+     * Generate method parameters for all the path params.
+     *
+     * @param pathParams the path param names to generate
+     * @return list of {@link ParameterSpec} that represent the method parameters
+     */
+    private List<ParameterSpec> methodPathParams(final Set<String> pathParams) {
+        return methodParams(pathParams, PathParam.class);
+    }
+
+    /**
+     * Generate code to add all parameters to the params map.
+     *
+     * @param params the param names to add to the map
+     * @return the {@link CodeBlock} that represents the generated code
+     */
+    private List<ParameterSpec> methodParams(final Collection<String> params, final Class<?> paramAnnotationClass) {
+        return params.stream().map(name ->
+                ParameterSpec
+                        .builder(String.class, name)
+                        .addAnnotation(AnnotationSpec.builder(paramAnnotationClass)
+                                .addMember(DEFAULT_ANNOTATION_PARAMETER, ANNOTATION_FORMAT, name)
+                                .build())
+                        .build()
+        ).collect(Collectors.toList());
+    }
+
+    /**
+     * Generates the Produces annotation for all the response mime types.
+     *
+     * @param responseMimeTypes generate annotations for each responseMimeType
+     * @return the list of {@link AnnotationSpec} that represent the produces annotations
+     */
+    private List<AnnotationSpec> annotationsForProduces(final Collection<MimeType> responseMimeTypes) {
+        final List<AnnotationSpec> specs = new ArrayList<>();
```

Comment:
I think it should be fine as final, we are just protecting the reference is not overwritten in this case